src/runtime/uv/timer.cpp

```diff
@@ -232,6 +240,34 @@ extern "C" LEAN_EXPORT lean_obj_res lean_uv_timer_stop(b_obj_arg obj, obj_arg /*
     }
 }
 
+/* Std.Internal.UV.Timer.cancel (timer : @& Timer) : IO Unit */
+extern "C" LEAN_EXPORT lean_obj_res lean_uv_timer_cancel(b_obj_arg obj, obj_arg /* w */) {
+    lean_uv_timer_object * timer = lean_to_uv_timer(obj);
+
+    // It's locking here to avoid changing the state during other operations.
+    event_loop_lock(&global_ev);
+
+    if (timer->m_state == TIMER_STATE_RUNNING && timer->m_promise != NULL) {
+        if (timer->m_repeating) {
+            lean_dec(timer->m_promise);
+            timer->m_promise = NULL;
+        } else {
+            uv_timer_stop(timer->m_uv_timer);
+
+            lean_dec(timer->m_promise);
+            timer->m_promise = NULL;
+            timer->m_state = TIMER_STATE_FINISHED;
+
+            lean_dec(obj);
```

Comment:
This `dec` seems a bit dubious given that we are borrowing the timer.

Reply:
If the loop is running, it increased the reference at some point, this lean_dec(obj) is just saying that it does not belong to libuv event loop anymore.